### PR TITLE
chore: Add service check workflow

### PR DESCRIPTION
**Pull request by sombrafam**
# Description

This code adds some GitHub workflow testing to make sure that the service is running.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation - N/A
- [x] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes - N/A
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -46,3 +55,36 @@ jobs:
...
+
+      - name: Install snap
+        run: |
+          SNAP_NAME=$(ls *.snap)
+          echo "Installing snap $SNAP_NAME"
+          sudo snap install --dangerous $SNAP_NAME
+
+      - name: Start the vault snap service
+        run: |
+          sudo snap start vault.service
```
Comment (saltiyazan) [inline .github/workflows/master.yml:77]: I think currently the name of the service is `vaultd` and we start it using `sudo snap start vault.vaultd`

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -16,14 +16,16 @@ jobs:
   build:
     name: Build snap
     runs-on: ubuntu-latest
+    outputs:
+      snapname: ${{ steps.upload-snap.outputs.artifact-id }}
```
Comment (saltiyazan) [inline .github/workflows/master.yml:20]: Are we using this anywhere?

**Review by saltiyazan — APPROVED** [review]
Looks good, thanks @sombrafam 
Added a minor comment.
I think you will have to sign the commit before merging.

File: .github/workflows/master.yml (hunk 2 above)
Comment (Kayra1) [reply .github/workflows/master.yml:20]: yes, in the Upload snap artifact step. Check [here](https://github.com/actions/upload-artifact?tab=readme-ov-file#outputs). The only problem I can see it the variable name here is `snapname` but the output required below looks at `outputs.snap`.

Comment (sombrafam) [reply .github/workflows/master.yml:20]: Not anymore. I was trying to pass along the name of the artifact but it wasn't working. Now I'm just listing the current directory.

Comment (sombrafam) [reply .github/workflows/master.yml]: Regarding the signature, see if is good now.

**Comment by saltiyazan** [comment]
Hi @sombrafam 
Should I merge?

**Comment by sombrafam** [comment]
Yes. Go ahead

**Pull request merged**